tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/NLog/LogsInjection/MappedDiagnosticsLogicalContextSetterProxy.cs

```diff
@@ -11,7 +11,9 @@ namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.NLog.LogsInjecti
     /// <summary>
     /// Duck type for MappedDiagnosticsLogicalContext in NLog 4.6+
     /// </summary>
+#pragma warning  disable DDSEAL001 // Types should be sealed where possible. Types used for duck-typing cannot be sealed.
     internal class MappedDiagnosticsLogicalContextSetterProxy
+#pragma warning restore DDSEAL001
```

Comment:
This is actually a (rare) `class` based duck type. We don't recommend these now (and can probably update to use `interface` instead, so just ignoring it for now